tests/components/nest/test_diagnostics.py

```diff
@@ -2,54 +2,45 @@
 
 from unittest.mock import patch
 
-from google_nest_sdm.device import Device
 from google_nest_sdm.exceptions import SubscriberException
+import pytest
 
-from homeassistant.components.nest import DOMAIN
 from homeassistant.config_entries import ConfigEntryState
-from homeassistant.setup import async_setup_component
 
-from .common import CONFIG, async_setup_sdm_platform, create_config_entry
+from .common import TEST_CONFIG_LEGACY
 
 from tests.components.diagnostics import get_diagnostics_for_config_entry
 
-THERMOSTAT_TYPE = "sdm.devices.types.THERMOSTAT"
 
-
-async def test_entry_diagnostics(hass, hass_client):
+async def test_entry_diagnostics(
```

Comment:
Note for reviewer: By default the nest tests using this setup_platform fixture run with multiple nest `NestTestConfig` objects. The legacy test below uses the same framework now and overrides with the new legacy nest test config.